client/fingerprint/vault.go

```diff
@@ -22,71 +23,90 @@ const (
 
 // VaultFingerprint is used to fingerprint for Vault
 type VaultFingerprint struct {
-	logger    log.Logger
-	client    *vapi.Client
-	lastState string
+	logger     log.Logger
+	clients    map[string]*vapi.Client
+	lastStates map[string]string
 }
 
 // NewVaultFingerprint is used to create a Vault fingerprint
 func NewVaultFingerprint(logger log.Logger) Fingerprint {
-	return &VaultFingerprint{logger: logger.Named("vault"), lastState: vaultUnavailable}
+	return &VaultFingerprint{
+		logger:     logger.Named("vault"),
+		clients:    map[string]*vapi.Client{},
+		lastStates: map[string]string{"default": vaultUnavailable},
+	}
 }
 
 func (f *VaultFingerprint) Fingerprint(req *FingerprintRequest, resp *FingerprintResponse) error {
-	config := req.Config
-
-	if config.VaultConfig == nil || !config.VaultConfig.IsEnabled() {
-		return nil
+	for _, cfg := range f.vaultConfigs(req) {
+		err := f.fingerprintImpl(cfg, resp)
+		if err != nil {
+			return err
+		}
```

Comment:
I imagine each fingerprint is fairly independent from each other, so it may be nice to accumulate errors from all fingerprints so operators can find and solve all problems at once instead of progressing through several attempts.

Reply:
Good call.